dev/new-rule/templates/rules-index.ejs

```diff
@@ -0,0 +1,7 @@
+'use strict';
+
+module.exports = {
```

Comment:
Let's put a comment in here saying this file is generated, and also add a `yarn update:rule-index` script to update it without having to add a new rule.

Reply:
Then you can also add a test to ensure that `yarn update:rule-index` results in "no changes".

Reply:
So I _could_ do this, but if you look at what `updateRulesIndex` is doing, it's using yeoman's generators to generate the file. If I change this to move away from yeoman's `fs`, it also opts us out of the yeoman runLoop, which governs when the file is updated/written. I don't think we want to do that, as it results in strange interleaving with the yeoman write process.

I'm happy to do a similar thing in a separate util. Up to you.

Reply:
Chatted offline about this with @scalvert. Let's make an issue to track this follow on work, it isn't blocking this from landing (it's just another nice improvement IMHO).